service/resolver/cri.go

```diff
@@ -136,18 +137,20 @@ func RegistryHostsFromCRIConfig(ctx context.Context, config Registry, credsFuncs
 			}
 
 			var (
-				transport = http.DefaultTransport.(*http.Transport).Clone()
-				client    = &http.Client{Transport: transport}
-				config    = config.Configs[u.Host]
+				rclient = rhttp.NewClient()
```

Comment:
Why not `client = rhttp.NewClient().StandardClient()` here?

Reply:
So that we can set `TLSClientConfig` on the `http.Transport` that is a field of `HTTPClient` (which is a field of `retryablehttp.Client`) using `getTLSConfig`.

Equivalent of doing:

```
client := rhttp.NewClient().StandardClient()
client.Transport.(*rhttp.RoundTripper).Client.HTTPClient.Transport.(*http.Transport).TLSConfig = getTLSConfig(*config.TLS)
```